programs/perpetuals/src/state/custody.rs

```diff
@@ -289,6 +358,121 @@ impl Custody {
 
         Ok(())
     }
+
+    pub fn get_collective_position(&self, side: Side) -> Result<Position> {
+        let stats = if side == Side::Long {
+            &self.long_positions
+        } else {
+            &self.short_positions
+        };
+        if stats.open_positions > 0 {
+            Ok(Position {
+                side,
+                price: math::checked_as_u64(math::checked_div(
+                    stats.weighted_leverage,
+                    stats.total_leverage,
+                )?)?,
+                size_usd: stats.size_usd,
+                collateral_usd: stats.collateral_usd,
+                unrealized_loss_usd: stats.cumulative_interest,
+                cumulative_interest_snapshot: stats.cumulative_interest_snapshot,
+                locked_amount: stats.locked_amount,
+                ..Position::default()
+            })
+        } else {
+            Ok(Position::default())
+        }
+    }
+
+    pub fn add_position(&mut self, position: &Position, curtime: i64) -> Result<()> {
+        // compute accumulated interest
+        let collective_position = self.get_collective_position(position.side)?;
+        let interest = self.get_interest_amount_usd(&collective_position, curtime)?;
```

Comment:
it should be almost near to 0 for new position.
if it is not, i think it will be double count when close position

Reply:
interest is calculated not for the position being added or removed, but rather for the cumulative long or short position. What happens here is we compute a combined outstanding interest for all opened positions and save it in `stats.cumulative_interest_usd` field. And we save current interest snapshot, so next time this function is called it will count only the interest that has been acquired since the previous call. But I agree that we should subtract position's interest when it is about to be closed

Reply:
updated the code